node/src/reactor/participating.rs

```diff
@@ -682,28 +682,80 @@ impl reactor::Reactor for Reactor {
+                                                        // to the cache.
+                                                        self.storage
+                                                            .deploy_cache
+                                                            .put(deploy_hash, Arc::downgrade(&arc));
+                                                        arc
+                                                    })
+                                                })
+                                        },
+                                    );
+
+                                match outcome {
+                                    Some(Ok(serialized)) => {
+                                        let message =
+                                            Message::new_get_response_raw_unchecked::<Deploy>(
+                                                SharedObject::shared(serialized),
+                                            );
+                                        return effect_builder
+                                            .send_message(sender, message)
+                                            .ignore();
+                                    }
+                                    Some(Err(err)) => {
+                                        error!(
+                                            err = display_error(&err),
+                                            "failed to create (shared) get-response"
+                                        );
+                                        return Effects::new();
+                                    }
+                                    None => {
+                                        debug!(%sender, %deploy_hash, "failed to get deploy (not found)");
+                                        return Effects::new();
+                                    }
                                 }
-                                None => {
-                                    debug!("failed to get {} for {}", deploy_hash, sender);
-                                    return Effects::new();
+                            } else {
+                                match self
+                                    .storage
+                                    .handle_legacy_direct_deploy_request(deploy_hash)
+                                {
+                                    // This functionality was moved out of the storage component and
+                                    // should be refactored ASAP.
+                                    Some(deploy) => {
+                                        match Message::new_get_response(&deploy) {
+                                            Ok(message) => {
+                                                if self.storage.mem_duplication_enabled() {
+                                                    todo!("update cache if enabled")
+                                                }
```

Comment:
Was this left in by mistake?  We're in an `else` branch here where `self.storage.mem_duplication_enabled()` is `false`.

Reply:
:( You're right.